src/AWS.Deploy.CLI/AWSUtilities.cs

```diff
@@ -24,63 +27,73 @@ public class AWSUtilities : IAWSUtilities
+            ICommandInputService commandInputService)
         {
             _toolInteractiveService = toolInteractiveService;
             _consoleUtilities = consoleUtilities;
             _directoryManager = directoryManager;
+            _commandInputService = commandInputService;
         }
 
         public async Task<AWSCredentials> ResolveAWSCredentials(string? profileName, string? lastUsedProfileName = null)
         {
             async Task<AWSCredentials> Resolve()
             {
-            var chain = new CredentialProfileStoreChain();
+                var chain = new CredentialProfileStoreChain();
 
                 if (!string.IsNullOrEmpty(profileName) && chain.TryGetAWSCredentials(profileName, out var profileCredentials) &&
                     // Skip checking CanLoadCredentials for AssumeRoleAWSCredentials because it might require an MFA token and the callback hasn't been setup yet.
                     (profileCredentials is AssumeRoleAWSCredentials || await CanLoadCredentials(profileCredentials)))
-            {
+                {
                     _toolInteractiveService.WriteLine($"Configuring AWS Credentials from Profile {profileName}.");
                     return profileCredentials;
                 }
 
-            if (!string.IsNullOrEmpty(lastUsedProfileName) &&
+                if (!string.IsNullOrEmpty(lastUsedProfileName) &&
                     chain.TryGetAWSCredentials(lastUsedProfileName, out var lastUsedCredentials) &&
                     await CanLoadCredentials(lastUsedCredentials))
-            {
-                _toolInteractiveService.WriteLine($"Configuring AWS Credentials with previous configured profile value {lastUsedProfileName}.");
+                {
+                    _toolInteractiveService.WriteLine($"Configuring AWS Credentials with previous configured profile value {lastUsedProfileName}.");
                     return lastUsedCredentials;
-            }
+                }
 
-            try
-            {
+                try
+                {
                     var fallbackCredentials = FallbackCredentialsFactory.GetCredentials();
 
                     if (await CanLoadCredentials(fallbackCredentials))
-                {
-                    _toolInteractiveService.WriteLine("Configuring AWS Credentials using AWS SDK credential search.");
+                    {
+                        _toolInteractiveService.WriteLine("Configuring AWS Credentials using AWS SDK credential search.");
                         return fallbackCredentials;
+                    }
+                }
+                catch (AmazonServiceException ex)
+                {
+                    // FallbackCredentialsFactory throws an exception if no credentials are found. Burying exception because if no credentials are found
+                    // we want to continue and ask the user to select a profile.
+                    if (_commandInputService.Diagnostics)
```

Comment:
Add a `WriteDiagnostics` method to IToolInteractiveServices and internal to that method check if diagnostics is enabled.

Reply:
Fixed